.changeset/nice-coats-lie.md

```diff
@@ -0,0 +1,7 @@
+---
+'astro': patch
+---
+
+`<style>` tags in components now follow the same rules as `<script>` tags, meaning that each page only includes the css of its imported components.
+
+Previously, some cases involving complex import trees could break Astro's analysis of imported components. Now, the analysis is a bit more robust and should handle more of those cases.
```

Comment:
```suggestion
Astro's static analysis to determine which `<style>` tags to bundle together just got a little smarter!

Each page now only includes the CSS of its imported components. Previously, some cases involving complex import trees could break Astro's analysis of imported components. Now, the analysis is a bit more robust and should handle more of those cases.
```
Hey @ottomated ! What do you think about matching the `<script>` version (adjust as necessary to be correct), rather than saying it works like scripts (which, might not be meaningful to a reader, depending on what they do or don't know about script bundling behaviour?)